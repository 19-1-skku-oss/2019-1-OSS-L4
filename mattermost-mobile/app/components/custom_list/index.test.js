// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {View} from 'react-native';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import CustomList, {FLATLIST, SECTIONLIST} from './index';

describe('CustomList', () => {
    const baseProps = {
        data: [{username: 'username_1'}, {username: 'username_2'}],
        listType: FLATLIST,
        loading: false,
        loadingComponent: (<View/>),
        onLoadMore: jest.fn(),
        onRowPress: jest.fn(),
        onRowSelect: null,
        renderItem: jest.fn(),
        listInitialSize: 0,
        listScrollRenderAheadDistance: 0,
        showSections: true,
        selectable: true,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot with FlatList', () => {
        const wrapper = shallow(
            <CustomList {...baseProps}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('FlatList')).toHaveLength(1);
    });

    test('should match snapshot with SectionList', () => {
        const props = {
            ...baseProps,
            listType: SECTIONLIST,
        };

        const wrapper = shallow(
            <CustomList {...props}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find('SectionList')).toHaveLength(1);
    });

    test('should match snapshot, renderSectionHeader', () => {
        const wrapper = shallow(
            <CustomList {...baseProps}/>
        );
        const section = {
            id: 'section_id',
        };
        expect(wrapper.instance().renderSectionHeader({section})).toMatchSnapshot();
    });

    test('should call props.renderItem on renderItem', () => {
        const props = {...baseProps};
        const wrapper = shallow(
            <CustomList {...props}/>
        );
        wrapper.instance().renderItem({item: {id: 'item_id', selected: true}, index: 0, section: null});
        expect(props.renderItem).toHaveBeenCalledTimes(1);
    });

    test('should match snapshot, renderSeparator', () => {
        const wrapper = shallow(
            <CustomList {...baseProps}/>
        );
        expect(wrapper.instance().renderSeparator()).toMatchSnapshot();
    });

    test('should match snapshot, renderFooter', () => {
        const props = {...baseProps};
        const wrapper = shallow(
            <CustomList {...props}/>
        );

        // should return null
        expect(wrapper.instance().renderFooter()).toMatchSnapshot();

        // should return element
        wrapper.setProps({loading: true});
        expect(wrapper.instance().renderFooter()).toMatchSnapshot();
    });
});
