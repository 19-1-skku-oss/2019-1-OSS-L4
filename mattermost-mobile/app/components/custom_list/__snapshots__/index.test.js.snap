// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomList should match snapshot with FlatList 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  data={
    Array [
      Object {
        "username": "username_1",
      },
      Object {
        "username": "username_2",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={15}
  keyExtractor={[Function]}
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
  maxToRenderPerBatch={16}
  numColumns={1}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  stickySectionHeadersEnabled={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`CustomList should match snapshot with SectionList 1`] = `
<SectionList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  data={Array []}
  disableVirtualization={false}
  extraData={false}
  horizontal={false}
  initialNumToRender={15}
  keyExtractor={[Function]}
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
  maxToRenderPerBatch={16}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={60}
  sections={
    Array [
      Object {
        "username": "username_1",
      },
      Object {
        "username": "username_2",
      },
    ]
  }
  stickySectionHeadersEnabled={false}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`CustomList should match snapshot, renderFooter 1`] = `null`;

exports[`CustomList should match snapshot, renderFooter 2`] = `<View />`;

exports[`CustomList should match snapshot, renderSectionHeader 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.07)",
        "paddingLeft": 10,
        "paddingVertical": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3d3c40",
          "fontWeight": "600",
        }
      }
    >
      section_id
    </Text>
  </View>
</View>
`;

exports[`CustomList should match snapshot, renderSeparator 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(61,60,64,0.1)",
      "flex": 1,
      "height": 1,
    }
  }
/>
`;
